### PR TITLE
Implement new designs for the editor and viewer's navbar

### PR DESCRIPTION
**Pull request by rishabhsaxena**
## Description
Implement new design for the  editor and viewer navbar

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/pages/Editor/EditorHeader.tsx [hunk 1]
```diff
@@ -189,78 +194,46 @@ export const EditorHeader = (props: EditorHeaderProps) => {
...
       <HeaderSection>
-        <Link to={APPLICATIONS_URL}>
+        <Link to={APPLICATIONS_URL} style={{ height: 24 }}>
           <AppsmithLogoImg
             src={AppsmithLogo}
             alt="Appsmith logo"
             className="t--appsmith-logo"
           />
         </Link>
+        {currentApplication ? (
```
Comment (hetunandu) [inline app/client/src/pages/Editor/EditorHeader.tsx:204]:
```suggestion
        {currentApplication && (
```

File: app/client/src/pages/Editor/EditorHeader.tsx [hunk 2]
```diff
@@ -189,78 +194,46 @@ export const EditorHeader = (props: EditorHeaderProps) => {
...
+              applicationList.filter((el) => el.id === applicationId).length > 0
+            }
+            onBlur={(value: string) =>
+              updateApplicationDispatch(applicationId || "", {
+                name: value,
+                currentApp: true,
+              })
+            }
+          />
+        ) : null}
```
Comment (hetunandu) [inline app/client/src/pages/Editor/EditorHeader.tsx:223]:
```suggestion
        )}
```

File: app/client/src/pages/Editor/EditorHeader.tsx (hunk 1 above)
Comment (hetunandu) [reply app/client/src/pages/Editor/EditorHeader.tsx:204]: This is not getting "Boxed" anymore

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/pages/Editor/EditorHeader.tsx (hunk 1 above)
Comment (rishabhsaxena) [reply app/client/src/pages/Editor/EditorHeader.tsx]: Should I remove the `Boxed` wrapper from the deploy button?

Comment (hetunandu) [reply app/client/src/pages/Editor/EditorHeader.tsx]: Looks like both need to be boxed. Can you confirm with Nikhil or Akash

**Review by hetunandu — DISMISSED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**